src/mesh/simplemesh.jl

```diff
@@ -44,3 +44,26 @@ list of points and [`FullTopology`](@ref).
 """
 Base.convert(::Type{<:SimpleMesh}, m::Mesh) =
   SimpleMesh(vertices(m), topology(m))
+
+
+"""
+    isinside(point, mesh)
+
+Tells whether a `point` is inside a surface `mesh`. Must be a surface mesh with only triangles as elements. Points that coincide with vertices of the mesh or lies on the face of a triangle return true.
+"""
+function isinside(testpoint::Point{3,T}, mesh::Mesh{3,T}) where T
+  if !(eltype(mesh) <: Triangle)
+    error("This function only works for surface meshes with triangles as elements.")
+  end
+  ex = testpoint .- extrema(mesh)
+  direction = ex[argmax(norm.(ex))]
+  r = Ray(testpoint, direction*2)
+  
+  intersects = false
+  for elem in mesh
+    if intersection(x -> x.type == NoIntersection ? false : true, r, elem)
+      intersects = !intersects
+    end
```

Comment:
Please try to use the do syntax:

```julia
if intersection(r, e) do I
  if type(I) == NoIntersection
    false
  else
    true
  end
end
```

Also, you probably can terminate the loop in the middle if the condition is met? You don't need to check all elements I guess? Can you please double check?

Reply:
You have to check all the elements, but I will adopt the do syntax.